### PR TITLE
Migrate node metrics to use new proc macro

### PR DESCRIPTION
**Pull request by jotare**
### Description
- Use new `nucliadb_procs::measure` macro for the current metrics

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Review by tarekziade — APPROVED** [review]
very nice improvement!

File: nucliadb_vectors/src/service/reader.rs [hunk 1]
```diff
@@ -65,44 +64,25 @@ impl Debug for VectorReaderService {
...
-            std::mem::drop(lock);
-
-            let metrics = metrics::get_metrics();
-            let took = time.elapsed().map(|i| i.as_secs_f64()).unwrap_or(f64::NAN);
-            let metric = request_time::RequestTimeKey::vectors("count".to_string());
-            metrics.record_request_time(metric, took);
-            debug!("Ending at {took} ms");
-
-            Ok(no_nodes)
+            self.index_count(&index)
```
Comment (lferran) [inline nucliadb_vectors/src/service/reader.rs:73]: Shouldn't this be a `Ok(self.index_count(&index)` ?

**Review by lferran — APPROVED** [review]
nice cleanup indeed! 👏🏼

File: nucliadb_vectors/src/service/reader.rs (hunk 1 above)
Comment (jotare) [reply nucliadb_vectors/src/service/reader.rs:73]: Index count already returns a Result, so no problem. Anyway, Rust will complain if it wasn't right

**Pull request merged**